featuretools/computational_backends/calculate_feature_matrix.py

```diff
@@ -172,30 +176,74 @@ def calculate_feature_matrix(features, cutoff_time=None, instance_ids=None,
     else:
         grouped = cutoff_time.groupby(cutoff_df_time_var, sort=True)
 
-    # if the backend is going to be verbose, don't make cutoff times verbose
-    if verbose and not backend_verbose:
-        iterator = make_tqdm_iterator(iterable=grouped,
-                                      total=len(grouped),
-                                      desc="Progress",
-                                      unit="cutoff time")
-    else:
-        iterator = grouped
-
-    feature_matrix = []
     backend = PandasBackend(entityset, features)
-    for _, group in iterator:
-        _feature_matrix = calculate_batch(features, group, approximate,
-                                          entityset, backend_verbose,
-                                          training_window, profile, verbose,
-                                          save_progress, backend,
-                                          no_unapproximated_aggs, cutoff_df_time_var,
-                                          target_time, pass_columns)
-        feature_matrix.append(_feature_matrix)
-        # Do a manual garbage collection in case objects from calculate_batch
-        # weren't collected automatically
-        gc.collect()
 
-    feature_matrix = pd.concat(feature_matrix)
+    # Determine shape of cutoff_time and calculate chunks
+    num_per_chunk = calc_num_per_chunk(chunk_size, cutoff_time.shape)
+
+    # initialize chunks
+    chunks = []
+    for _, group in grouped:
+        if group.shape[0] > num_per_chunk:
+            for i in range(0, group.shape[0], num_per_chunk):
+                chunks.append((_, group.iloc[i:i + num_per_chunk]))
+        else:
+            chunks.append((_, group))
```

Comment:
Can't you just make chunks a list of [group] instead of [(_, group)], and then just parallelize calculate_batch directly instead of having the calculate_chunk wrapper function? It seems like calculate_chunk doesn't do anything